src/commands/leaderboard.ts

```diff
@@ -0,0 +1,72 @@
+import { CommandInteraction } from "discord.js";
+import { SlashCommandBuilder } from "@discordjs/builders";
+import { RequestError, getLeaderboard, Language, LeaderboardPosition } from "../codewars";
+import { table, TableUserConfig } from "table";
+import { findLanguage, checkBotPlayground } from "../common";
+export { languageAutocomplete as autocomplete } from "../common";
+
+const tableConfig: TableUserConfig = { 
+  drawHorizontalLine: (lineIndex: number, rowCount: number) => lineIndex < 2 || lineIndex === rowCount,
+  columns: [
+    { alignment: 'right' },
+    { alignment: 'left' },
+    { alignment: 'right' },
+    { alignment: 'left' }
+  ],
+};
+
+// leaderboard
+export const data = async () =>
+  new SlashCommandBuilder()
+    .setName("leaderboard")
+    .setDescription("Show the language score leaderboards")
+    .addStringOption((option) =>
+      option
+        .setName("language")
+        .setDescription("The programming language to show leaderboard for")
+        .setAutocomplete(true)
+    )
+    .addIntegerOption((option) =>
+      option.setName("startposition")
+        .setDescription("The top shown position (default 1)")
+        // limits seem not to be checked
+        .setMinValue(1)
+    )
+    .addIntegerOption((option) =>
+      option.setName("limit").setDescription("The number of shown users (default 10, range 1 - 25)")
+      .setMinValue(1)
+      // cut off at 25 because of Discord message length limit, 31 would fit
+      .setMaxValue(25)
+    )
+    .addBooleanOption((option) =>
+      option.setName("ephemeral").setDescription("Don't show leaderboard to others")
+    )
+    .toJSON();
+
+function format(
+  lang: Language | null,
+  leaderboard: LeaderboardPosition[]
+) {
+  const data = [['#','username','score','rank'], ...leaderboard.map(u=>[u.position, u.username, u.score, getRank(u.rank)])];
+  return `:trophy: **${lang?.name ?? "Overall"} leaderboard** :medal:\n`+"```\n"+table(data, tableConfig)+"\n```";
+}
+function getRank(rank: number) {
+  if(rank > 0) return rank + " dan";
```

Comment:
Please format the code with Prettier. https://github.com/codewars/discord-bot#code-style

How did you make this PR? Running `npm install` should install the `pre-commit` hook to autoformat the changed code.

Reply:
I had to run `npx pretty-quick` manually in the end.